From 16b1660f503523d880d43f2c34da5b4273c06db9 Mon Sep 17 00:00:00 2001
From: Sean McCreary <mccreary@mcwest.org>
Date: Mon, 9 Apr 2018 16:01:37 -0600
Subject: [PATCH 210/249] heimdall: Add reminder to enable OEM unlock

Disable reminder by default for all devices using
recovery_install_heimdall.md, except:

	a5y17lte
	a7y17lte
	gts210ltexx
	gts210wifi
	gts210vewifi
	gts28vewifi
	herolte
	hero2lte
	starlte
	star2lte

Change-Id: I19099d1584eda601459770d57a028fdc7bb3018d
---
 _data/devices/chagallwifi.yml                    |  1 +
 _data/devices/d2att.yml                          |  1 +
 _data/devices/d2spr.yml                          |  1 +
 _data/devices/d2tmo.yml                          |  1 +
 _data/devices/d2vzw.yml                          |  1 +
 _data/devices/espresso3g.yml                     |  1 +
 _data/devices/espressowifi.yml                   |  1 +
 _data/devices/gtelwifiue.yml                     |  1 +
 _data/devices/gtesqltespr.yml                    |  1 +
 _data/devices/ha3g.yml                           |  1 +
 _data/devices/hlte.yml                           |  1 +
 _data/devices/hltetmo.yml                        |  1 +
 _data/devices/i9100.yml                          |  1 +
 _data/devices/i9300.yml                          |  1 +
 _data/devices/i9305.yml                          |  1 +
 _data/devices/jfltevzw.yml                       |  1 +
 _data/devices/jfltexx.yml                        |  1 +
 _data/devices/k3gxx.yml                          |  1 +
 _data/devices/kccat6.yml                         |  1 +
 _data/devices/klimtwifi.yml                      |  1 +
 _data/devices/klte.yml                           |  1 +
 _data/devices/klteactivexx.yml                   |  1 +
 _data/devices/kltechn.yml                        |  1 +
 _data/devices/kltechnduo.yml                     |  1 +
 _data/devices/klteduos.yml                       |  1 +
 _data/devices/kltedv.yml                         |  1 +
 _data/devices/kltekdi.yml                        |  1 +
 _data/devices/kltekor.yml                        |  1 +
 _data/devices/kltesprsports.yml                  |  1 +
 _data/devices/ks01lte.yml                        |  1 +
 _data/devices/lentislte.yml                      |  1 +
 _data/devices/lt02ltespr.yml                     |  1 +
 _data/devices/lt03lte.yml                        |  1 +
 _data/devices/mondrianwifi.yml                   |  1 +
 _data/devices/n1awifi.yml                        |  1 +
 _data/devices/n2awifi.yml                        |  1 +
 _data/devices/n5100.yml                          |  1 +
 _data/devices/n5110.yml                          |  1 +
 _data/devices/n5120.yml                          |  1 +
 _data/devices/serrano3gxx.yml                    |  1 +
 _data/devices/serranodsdd.yml                    |  1 +
 _data/devices/serranoltexx.yml                   |  1 +
 _data/devices/t0lte.yml                          |  1 +
 _data/devices/v1awifi.yml                        |  1 +
 _data/devices/zerofltexx.yml                     |  1 +
 _data/devices/zeroltexx.yml                      |  1 +
 _includes/templates/recovery_install_heimdall.md | 15 +++++++++------
 47 files changed, 55 insertions(+), 6 deletions(-)

diff --git a/_data/devices/chagallwifi.yml b/_data/devices/chagallwifi.yml
index aec681f..02f2017 100644
--- a/_data/devices/chagallwifi.yml
+++ b/_data/devices/chagallwifi.yml
@@ -25,6 +25,7 @@ network:
 - {tech: 2G, bands: '' }
 - {tech: 3G, bands: '' }
 - {tech: 4G, bands: '' }
+no_oem_unlock_switch: true
 peripherals:
 - SM-T800: [Accelerometer, Compass, GPS, Gyroscope]
 - Unsupported: [Fingerprint reader]
diff --git a/_data/devices/d2att.yml b/_data/devices/d2att.yml
index e3cf298..e28b436 100644
--- a/_data/devices/d2att.yml
+++ b/_data/devices/d2att.yml
@@ -25,6 +25,7 @@ network:
 - {tech: 2G, bands: '850 900 1800 1900 MHz GSM'}
 - {tech: 3G, bands: '850 1900 2100 MHz HSPA+'}
 - {tech: 4G, bands: '13(700)'}
+no_oem_unlock_switch: true
 peripherals: [Proximity sensor, Light sensor, Accelerometer, Gyroscope, Compass, A-GPS, NFC]
 ram: 2 GB
 recovery_boot: With the device powered off, hold <kbd>Volume Up</kbd> + <kbd>Home</kbd> + <kbd>Power</kbd>.
diff --git a/_data/devices/d2spr.yml b/_data/devices/d2spr.yml
index b9854db..cc9f47d 100644
--- a/_data/devices/d2spr.yml
+++ b/_data/devices/d2spr.yml
@@ -25,6 +25,7 @@ network:
 - {tech: 2G, bands: '850 900 1800 1900 MHz GSM' }
 - {tech: 3G, bands: '850 1900 2100 MHz HSPA+' }
 - {tech: 4G, bands: '1900 MHz' }
+no_oem_unlock_switch: true
 peripherals: [Proximity sensor, Light sensor, Accelerometer, Gyroscope, Compass, A-GPS, NFC]
 ram: 2 GB
 recovery_boot: With the device powered off, hold <kbd>Volume Up</kbd> + <kbd>Home</kbd> + <kbd>Power</kbd>.
diff --git a/_data/devices/d2tmo.yml b/_data/devices/d2tmo.yml
index fe869f5..24db36c 100644
--- a/_data/devices/d2tmo.yml
+++ b/_data/devices/d2tmo.yml
@@ -25,6 +25,7 @@ network:
 - {tech: 2G, bands: '850 900 1800 1900 MHz GSM'}
 - {tech: 3G, bands: '850 1700 1900 2100 MHz HSPA+'}
 - {tech: 4G, bands: ''}
+no_oem_unlock_switch: true
 peripherals: [Proximity sensor, Light sensor, Accelerometer, Gyroscope, Compass, A-GPS, NFC]
 ram: 2 GB
 recovery_boot: With the device powered off, hold <kbd>Volume Up</kbd> + <kbd>Home</kbd> + <kbd>Power</kbd>.
diff --git a/_data/devices/d2vzw.yml b/_data/devices/d2vzw.yml
index 339080b..81c454a 100644
--- a/_data/devices/d2vzw.yml
+++ b/_data/devices/d2vzw.yml
@@ -25,6 +25,7 @@ network:
 - {tech: 2G, bands: '800 1900 MHz CDMA' }
 - {tech: 3G, bands: 'CDMA2000 1xEV-DO' }
 - {tech: 4G, bands: '13(700)' }
+no_oem_unlock_switch: true
 peripherals: [Proximity sensor, Light sensor, Accelerometer, Gyroscope, Compass, A-GPS, NFC]
 ram: 2 GB
 recovery_boot: With the device powered off, hold <kbd>Volume Up</kbd> + <kbd>Home</kbd> + <kbd>Power</kbd>.
diff --git a/_data/devices/espresso3g.yml b/_data/devices/espresso3g.yml
index 58f7167..7a7720a 100644
--- a/_data/devices/espresso3g.yml
+++ b/_data/devices/espresso3g.yml
@@ -27,6 +27,7 @@ network:
 - {tech: 2G, bands: '850 900 1800 1900 MHz' }
 - {tech: 3G, bands: '' }
 - {tech: 4G, bands: '' }
+no_oem_unlock_switch: true
 peripherals: [Proximity sensor, Light sensor, Accelerometer, Gyroscope, Compass, GPS]
 ram: 1 GB
 recovery_boot: With the device powered off, hold <kbd>Volume Up</kbd> + <kbd>Power</kbd>.
diff --git a/_data/devices/espressowifi.yml b/_data/devices/espressowifi.yml
index 99905e1..277551e 100644
--- a/_data/devices/espressowifi.yml
+++ b/_data/devices/espressowifi.yml
@@ -27,6 +27,7 @@ network:
 - {tech: 2G, bands: '' }
 - {tech: 3G, bands: '' }
 - {tech: 4G, bands: '' }
+no_oem_unlock_switch: true
 peripherals: [Proximity sensor, Light sensor, Accelerometer, Gyroscope, Compass, GPS]
 ram: 1 GB
 recovery_boot: With the device powered off, hold <kbd>Volume Up</kbd> + <kbd>Power</kbd>.
diff --git a/_data/devices/gtelwifiue.yml b/_data/devices/gtelwifiue.yml
index 73cd6cf..0ae2c41 100644
--- a/_data/devices/gtelwifiue.yml
+++ b/_data/devices/gtelwifiue.yml
@@ -25,6 +25,7 @@ network:
 - {tech: 2G, bands: '' }
 - {tech: 3G, bands: '' }
 - {tech: 4G, bands: '' }
+no_oem_unlock_switch: true
 peripherals: [GPS, GLONASS, USB OTG, Accelerometer, Compass]
 ram: 1.5 GB
 recovery_boot: With the device powered off, hold <kbd>Home</kbd> + <kbd>Volume Up</kbd> + <kbd>Power</kbd>.
diff --git a/_data/devices/gtesqltespr.yml b/_data/devices/gtesqltespr.yml
index a2c8cd4..4adca11 100644
--- a/_data/devices/gtesqltespr.yml
+++ b/_data/devices/gtesqltespr.yml
@@ -25,6 +25,7 @@ network:
 - {tech: 2G, bands: '800 1900 MHz CDMA' }
 - {tech: 3G, bands: 'CDMA2000 1xEV-DO CDMA2000 1xRTT' }
 - {tech: 4G, bands: '800 1900 2500 MHz LTE' }
+no_oem_unlock_switch: true
 peripherals: [GPS, GLONASS, USB OTG, Accelerometer, Compass]
 ram: 1.5 GB
 recovery_boot: With the device powered off, hold <kbd>Home</kbd> + <kbd>Volume Up</kbd> + <kbd>Power</kbd>.
diff --git a/_data/devices/ha3g.yml b/_data/devices/ha3g.yml
index 0d89782..7a7e3b8 100644
--- a/_data/devices/ha3g.yml
+++ b/_data/devices/ha3g.yml
@@ -25,6 +25,7 @@ network:
 - {tech: 2G, bands: '850 900 1800 1900 MHz GSM' }
 - {tech: 3G, bands: '850 900 1900 2100 MHz HSDPA' }
 - {tech: 4G, bands: '' }
+no_oem_unlock_switch: true
 peripherals: [Accelerometer, Gyroscope, Proximity sensor, Compass, Barometer, Thermometer, Hygrometer, NFC, GPS, GLONASS]
 ram: 3 GB
 recovery_boot: With the device powered off, hold <kbd>Volume Up</kbd> + <kbd>Home</kbd> + <kbd>Power</kbd>.
diff --git a/_data/devices/hlte.yml b/_data/devices/hlte.yml
index 41c9a29..4699fe3 100644
--- a/_data/devices/hlte.yml
+++ b/_data/devices/hlte.yml
@@ -25,6 +25,7 @@ network:
 - {tech: 2G, bands: '850 900 1800 1900 MHz GSM' }
 - {tech: 3G, bands: '850 900 1900 2100 MHz HSDPA (N9005), 1900 2100 MHz HSPDA (N900P), 900 1900 MHz CDMA (N900P)' }
 - {tech: 4G, bands: '1(2100) 3(1800) 5(850) 7(2600) 8(900) 20(800) (N9005), 25(1900) (N900P)' }
+no_oem_unlock_switch: true
 peripherals: [Accelerometer, Gyroscope, Proximity sensor, Compass, Barometer, Thermometer, Hygrometer, NFC, GPS, GLONASS]
 ram: 3 GB
 recovery_boot: With the device powered off, hold <kbd>Volume Up</kbd> + <kbd>Home</kbd> + <kbd>Power</kbd>. When the blue text appears, release the buttons.
diff --git a/_data/devices/hltetmo.yml b/_data/devices/hltetmo.yml
index 933d185..83071ba 100644
--- a/_data/devices/hltetmo.yml
+++ b/_data/devices/hltetmo.yml
@@ -25,6 +25,7 @@ network:
 - {tech: 2G, bands: '850 900 1800 1900 MHz GSM' }
 - {tech: 3G, bands: '850 900 1900 2100 MHz HSDPA (N900T/V/W8), 800 1900 MHz CDMA (N900V)' }
 - {tech: 4G, bands: '4(1700/2100) 17(700) (N900T), 4(1700/2100) 13(700) (N900V), 2(1900) 4(1700/2100) 17(700) (N900W8)' }
+no_oem_unlock_switch: true
 peripherals: [Accelerometer, Gyroscope, Proximity sensor, Compass, Barometer, Thermometer, Hygrometer, NFC, GPS, GLONASS]
 ram: 3 GB
 recovery_boot: With the device powered off, hold <kbd>Volume Up</kbd> + <kbd>Home</kbd> + <kbd>Power</kbd>. When the blue text appears, release the buttons.
diff --git a/_data/devices/i9100.yml b/_data/devices/i9100.yml
index d197a75..aece72b 100644
--- a/_data/devices/i9100.yml
+++ b/_data/devices/i9100.yml
@@ -25,6 +25,7 @@ network:
 - {tech: 2G, bands: '850 900 1800 1900 MHz GSM'}
 - {tech: 3G, bands: '850 900 1900 2100 MHz HSPA+'}
 - {tech: 4G, bands: ''}
+no_oem_unlock_switch: true
 peripherals: None
 ram: 1 GB
 recovery_boot: With the device powered off, hold <kbd>Volume Up</kbd> + <kbd>Home</kbd> + <kbd>Power</kbd>.
diff --git a/_data/devices/i9300.yml b/_data/devices/i9300.yml
index 1d64bf1..c01f606 100644
--- a/_data/devices/i9300.yml
+++ b/_data/devices/i9300.yml
@@ -25,6 +25,7 @@ network:
 - {tech: 2G, bands: '850 900 1800 1900 MHz GSM'}
 - {tech: 3G, bands: '850 900 1900 2100 MHz HSDPA'}
 - {tech: 4G, bands: ''}
+no_oem_unlock_switch: true
 peripherals: [Accelerometer, Gyroscope, Proximity sensor, Compass, GPS, NFC, Barometer]
 ram: 1 GB
 recovery_boot: With the device powered off, hold <kbd>Home</kbd> + <kbd>Volume Up</kbd> + <kbd>Power</kbd>.
diff --git a/_data/devices/i9305.yml b/_data/devices/i9305.yml
index cd041fa..0faa36e 100644
--- a/_data/devices/i9305.yml
+++ b/_data/devices/i9305.yml
@@ -25,6 +25,7 @@ network:
 - {tech: 2G, bands: '850 900 1800 1900 MHz GSM'}
 - {tech: 3G, bands: '850 900 2100 MHz UMTS/HSPA+'}
 - {tech: 4G, bands: '800 1800 2600 MHz (GT-I9305), 900 1800 2600 MHz (GT-I9305N), 1800 2600 MHz (GT-I9305T)'}
+no_oem_unlock_switch: true
 peripherals: [Accelerometer, Gyroscope, Proximity sensor, Compass, GPS, NFC]
 ram: 2 GB
 recovery_boot: With the device powered off, hold <kbd>Home</kbd> + <kbd>Volume Up</kbd> + <kbd>Power</kbd>.
diff --git a/_data/devices/jfltevzw.yml b/_data/devices/jfltevzw.yml
index af676a4..a74a3dd 100644
--- a/_data/devices/jfltevzw.yml
+++ b/_data/devices/jfltevzw.yml
@@ -27,6 +27,7 @@ network:
 - {tech: 2G, bands: '850 900 1800 1900 MHz GSM, 800 1900 MHz CDMA' }
 - {tech: 3G, bands: '850 900 1900 2100 MHz HSPA+ / UMTS, CDMA2000 1xEV-DO' }
 - {tech: 4G, bands: '1700/2100 700 MHz LTE' }
+no_oem_unlock_switch: true
 peripherals: [GPS, GLONASS, NFC, Infrared sensor, USB OTG, MHL 2.0, MHL HDMI, Accelerometer, Barometer,
   Gesture sensor, Gyroscope, Hall effect sensor, Hygrometer, Proximity sensor, Light sensor, Thermometer]
 ram: 2 GB
diff --git a/_data/devices/jfltexx.yml b/_data/devices/jfltexx.yml
index 31e6f20..1738d59 100644
--- a/_data/devices/jfltexx.yml
+++ b/_data/devices/jfltexx.yml
@@ -26,6 +26,7 @@ network:
 - {tech: 2G, bands: '850 900 1800 1900 MHz GSM' }
 - {tech: 3G, bands: '850 900 1900 2100 MHz HSPA+' }
 - {tech: 4G, bands: '800 850 900 1800 2100 2600 MHz LTE' }
+no_oem_unlock_switch: true
 peripherals: [GPS, GLONASS, NFC, Infrared sensor, USB OTG, MHL 2.0, MHL HDMI, Accelerometer, Barometer,
   Gesture sensor, Gyroscope, Hall effect sensor, Hygrometer, Proximity sensor, Light sensor, Thermometer]
 ram: 2 GB
diff --git a/_data/devices/k3gxx.yml b/_data/devices/k3gxx.yml
index ba9d087..af14cd3 100644
--- a/_data/devices/k3gxx.yml
+++ b/_data/devices/k3gxx.yml
@@ -25,6 +25,7 @@ network:
 - {tech: 2G, bands: '850 900 1800 1900 MHz GSM' }
 - {tech: 3G, bands: '850 900 1900 2100 MHz HSDPA' }
 - {tech: 4G, bands: '' }
+no_oem_unlock_switch: true
 peripherals: [Accelerometer, Gyroscope, Proximity sensor, Compass, Barometer, Thermometer, Hygrometer, NFC, GPS, GLONASS, Fingerprint reader]
 ram: 2 GB
 recovery_boot: With the device powered off, hold <kbd>Volume Up</kbd> + <kbd>Home</kbd> + <kbd>Power</kbd>.
diff --git a/_data/devices/kccat6.yml b/_data/devices/kccat6.yml
index e345992..67230b8 100644
--- a/_data/devices/kccat6.yml
+++ b/_data/devices/kccat6.yml
@@ -24,6 +24,7 @@ network:
 - {tech: 2G, bands: '850 900 1800 1900 MHz GSM' }
 - {tech: 3G, bands: '850 900 1900 2100 MHz HSDPA' }
 - {tech: 4G, bands: '1(2100) 2(1900) 3(1800) 5(850) 7(2600) 8(900) 20(800)' }
+no_oem_unlock_switch: true
 peripherals: [Fingerprint reader, Heart rate sensor, Motion Coprocessor, Accelerometer, Gesture sensor,
   Gyroscope, Proximity sensor, Compass, Barometer, Hall effect sensor, RGB ambient light, Infrared sensor]
 ram: 2 GB
diff --git a/_data/devices/klimtwifi.yml b/_data/devices/klimtwifi.yml
index 0a8ff8a..ac829df 100644
--- a/_data/devices/klimtwifi.yml
+++ b/_data/devices/klimtwifi.yml
@@ -25,6 +25,7 @@ network:
 - {tech: 2G, bands: '' }
 - {tech: 3G, bands: '' }
 - {tech: 4G, bands: '' }
+no_oem_unlock_switch: true
 peripherals:
 - SM-T700: [Accelerometer, Compass, GPS, Gyroscope]
 - Unsupported: [Fingerprint reader]
diff --git a/_data/devices/klte.yml b/_data/devices/klte.yml
index c4daa82..f0bc98e 100644
--- a/_data/devices/klte.yml
+++ b/_data/devices/klte.yml
@@ -25,6 +25,7 @@ network:
 - {tech: 2G, bands: '850 900 1800 1900 MHz GSM, 800 1900MHz 1xRTT (G900R4/G900R7/G900V/S902L)' }
 - {tech: 3G, bands: '850 900 1900 2100 MHz HSDPA (G900F/G900V), 850 900 1700 1900 2100 MHz HSDPA (G900M), 850 1700 1900 2100 MHz HSDPA (G900T), 800 1900MHz EVDO (G900R4/G900R7/G900V/S902L)' }
 - {tech: 4G, bands: '1(2100) 2(1900) 3(1800) 5(850) 7(2600) 8(900) 20(800) (G900F), 1(2100) 2(1900) 4(1700/2100) 5(850) 7(2600) 17(700) (G900M), 1(2100) 2(1900) 3(1800) 4(1700/2100) 5(850) 7(2600) 8(900) 17(700) (G900T), 2(1900) 4(1700/2100) 5(850) 12(700) 17(700) (G900R4), 2(1900) 4(1700/2100) 12(700) (G900R7), 4(1700/2100) 13(700) (G900V/S902L)' }
+no_oem_unlock_switch: true
 peripherals: [Fingerprint reader, Heart rate sensor, Motion Coprocessor, Accelerometer, Gesture sensor,
   Gyroscope, Proximity sensor, Compass, Barometer, Hall effect sensor, RGB ambient light, Infrared sensor]
 ram: 2 GB
diff --git a/_data/devices/klteactivexx.yml b/_data/devices/klteactivexx.yml
index a64b191..5a261ec 100644
--- a/_data/devices/klteactivexx.yml
+++ b/_data/devices/klteactivexx.yml
@@ -24,6 +24,7 @@ network:
 - {tech: 2G, bands: '850 900 1800 1900 MHz GSM' }
 - {tech: 3G, bands: '850 900 1900 2100 MHz UMTS' }
 - {tech: 4G, bands: '(1)2100 (2)1900 3(1800) 5(850) (7)2600 (8)900 (20)800 LTE' }
+no_oem_unlock_switch: true
 peripherals: [Heart rate sensor, Accelerometer, Gesture sensor, Gyroscope, Proximity sensor, Compass, Barometer, RGB ambient light, Infrared sensor]
 ram: 2 GB
 recovery_boot: With the device powered off, hold <kbd>Volume Up</kbd> + <kbd>Home</kbd> + <kbd>Power</kbd>. When the blue text appears, release the buttons.
diff --git a/_data/devices/kltechn.yml b/_data/devices/kltechn.yml
index d486ea5..86dbdbc 100644
--- a/_data/devices/kltechn.yml
+++ b/_data/devices/kltechn.yml
@@ -25,6 +25,7 @@ network:
 - {tech: 2G, bands: '' }
 - {tech: 3G, bands: '' }
 - {tech: 4G, bands: '' }
+no_oem_unlock_switch: true
 peripherals: [Fingerprint reader, Heart rate sensor, Motion Coprocessor, Accelerometer, Gesture sensor,
   Gyroscope, Proximity sensor, Compass, Barometer, Hall effect sensor, RGB ambient light, Infrared sensor]
 ram: 2 GB
diff --git a/_data/devices/kltechnduo.yml b/_data/devices/kltechnduo.yml
index 5a2b2e9..d175d22 100644
--- a/_data/devices/kltechnduo.yml
+++ b/_data/devices/kltechnduo.yml
@@ -26,6 +26,7 @@ network:
 - {tech: 2G, bands: '' }
 - {tech: 3G, bands: '' }
 - {tech: 4G, bands: '' }
+no_oem_unlock_switch: true
 peripherals: [Fingerprint reader, Heart rate sensor, Motion Coprocessor, Accelerometer, Gesture sensor,
   Gyroscope, Proximity sensor, Compass, Barometer, Hall effect sensor, RGB ambient light, Infrared sensor]
 ram: 2 GB
diff --git a/_data/devices/klteduos.yml b/_data/devices/klteduos.yml
index 779add8..60f4b3f 100644
--- a/_data/devices/klteduos.yml
+++ b/_data/devices/klteduos.yml
@@ -25,6 +25,7 @@ network:
 - {tech: 2G, bands: '' }
 - {tech: 3G, bands: '' }
 - {tech: 4G, bands: '' }
+no_oem_unlock_switch: true
 peripherals: [Fingerprint reader, Heart rate sensor, Motion Coprocessor, Accelerometer, Gesture sensor,
   Gyroscope, Proximity sensor, Compass, Barometer, Hall effect sensor, RGB ambient light, Infrared sensor]
 ram: 2 GB
diff --git a/_data/devices/kltedv.yml b/_data/devices/kltedv.yml
index a6d85bc..6739d12 100644
--- a/_data/devices/kltedv.yml
+++ b/_data/devices/kltedv.yml
@@ -26,6 +26,7 @@ network:
 - {tech: 2G, bands: '' }
 - {tech: 3G, bands: '' }
 - {tech: 4G, bands: '' }
+no_oem_unlock_switch: true
 peripherals: [Fingerprint reader, Heart rate sensor, Motion Coprocessor, Accelerometer, Gesture sensor,
   Gyroscope, Proximity sensor, Compass, Barometer, Hall effect sensor, RGB ambient light, Infrared sensor]
 ram: 2 GB
diff --git a/_data/devices/kltekdi.yml b/_data/devices/kltekdi.yml
index b1f6d83..dc975d7 100644
--- a/_data/devices/kltekdi.yml
+++ b/_data/devices/kltekdi.yml
@@ -25,6 +25,7 @@ network:
 - {tech: 2G, bands: '' }
 - {tech: 3G, bands: '' }
 - {tech: 4G, bands: '' }
+no_oem_unlock_switch: true
 peripherals: [Fingerprint reader, Heart rate sensor, Motion Coprocessor, Accelerometer, Gesture sensor,
   Gyroscope, Proximity sensor, Compass, Barometer, Hall effect sensor, RGB ambient light, Infrared sensor]
 ram: 2 GB
diff --git a/_data/devices/kltekor.yml b/_data/devices/kltekor.yml
index 24420d4..11286db 100644
--- a/_data/devices/kltekor.yml
+++ b/_data/devices/kltekor.yml
@@ -26,6 +26,7 @@ network:
 - {tech: 2G, bands: '' }
 - {tech: 3G, bands: '' }
 - {tech: 4G, bands: '' }
+no_oem_unlock_switch: true
 peripherals: [Fingerprint reader, Heart rate sensor, Motion Coprocessor, Accelerometer, Gesture sensor, Gyroscope,
   Proximity sensor, Compass, Barometer, Hall effect sensor, RGB ambient light, Infrared sensor]
 ram: 2 GB
diff --git a/_data/devices/kltesprsports.yml b/_data/devices/kltesprsports.yml
index ded1b99..6696056 100644
--- a/_data/devices/kltesprsports.yml
+++ b/_data/devices/kltesprsports.yml
@@ -26,6 +26,7 @@ network:
 - {tech: 2G, bands: '850 900 1800 1900 MHz GSM, 800 1900 MHz CDMA' }
 - {tech: 3G, bands: '850 900 1900 2100 MHz UMTS, 800 1900 MHz EvDO' }
 - {tech: 4G, bands: '25(1900) 26(850) 41(2500) LTE' }
+no_oem_unlock_switch: true
 peripherals: [Heart rate sensor, Motion Coprocessor, Accelerometer, Gesture sensor, Gyroscope, Proximity sensor, Compass, Barometer, RGB ambient light, Infrared sensor]
 ram: 2 GB
 recovery_boot: With the device powered off, hold <kbd>Volume Up</kbd> + <kbd>Home</kbd> + <kbd>Power</kbd>. When the blue text appears, release the buttons.
diff --git a/_data/devices/ks01lte.yml b/_data/devices/ks01lte.yml
index a15c91b..9d12573 100644
--- a/_data/devices/ks01lte.yml
+++ b/_data/devices/ks01lte.yml
@@ -25,6 +25,7 @@ network:
 - {tech: 2G, bands: '850 900 1800 1900 MHz GSM' }
 - {tech: 3G, bands: '850 900 1900 2100 MHz HSDPA' }
 - {tech: 4G, bands: '1(2100) 3(1800) 5(850) 7(2600) 8(900) 20(800) LTE' }
+no_oem_unlock_switch: true
 peripherals: [GPS, GLONASS, NFC, Infrared sensor, USB OTG, MHL 2.0, MHL HDMI, Accelerometer, Barometer, Gesture sensor,
   Gyroscope, Hall effect sensor, Hygrometer, Proximity sensor, Light sensor, Thermometer]
 ram: 2 GB
diff --git a/_data/devices/lentislte.yml b/_data/devices/lentislte.yml
index 87d4e02..e5f742f 100644
--- a/_data/devices/lentislte.yml
+++ b/_data/devices/lentislte.yml
@@ -24,6 +24,7 @@ network:
 - {tech: 2G, bands: '900 1800 1900 MHz GSM' }
 - {tech: 3G, bands: '850 1900 2100 MHz HSDPA' }
 - {tech: 4G, bands: '1(2100) 3(1800) 5(850) 7(2600) 17(700)' }
+no_oem_unlock_switch: true
 peripherals: [Fingerprint reader, Heart rate sensor, Motion Coprocessor, Accelerometer, Gesture sensor, Gyroscope, Proximity sensor,
   Compass, Barometer, Hall effect sensor, RGB ambient light, IR blaster]
 ram: 3 GB
diff --git a/_data/devices/lt02ltespr.yml b/_data/devices/lt02ltespr.yml
index b2c819f..453f397 100644
--- a/_data/devices/lt02ltespr.yml
+++ b/_data/devices/lt02ltespr.yml
@@ -25,6 +25,7 @@ network:
 - {tech: 2G, bands: '800 1900 MHz CDMA' }
 - {tech: 3G, bands: 'CDMA2000 1xEV-DO CDMA2000 1xRTT' }
 - {tech: 4G, bands: '1900 MHz LTE' }
+no_oem_unlock_switch: true
 peripherals: [IR blaster, Accelerometer, Light sensor, Compass]
 ram: 1.5 GB
 recovery_boot: With the device powered off, hold <kbd>Home</kbd> + <kbd>Volume Up</kbd> + <kbd>Power</kbd>.
diff --git a/_data/devices/lt03lte.yml b/_data/devices/lt03lte.yml
index ad60972..036d5bd 100644
--- a/_data/devices/lt03lte.yml
+++ b/_data/devices/lt03lte.yml
@@ -22,6 +22,7 @@ network:
 - {tech: 2G, bands: '' }
 - {tech: 3G, bands: '' }
 - {tech: 4G, bands: '' }
+no_oem_unlock_switch: true
 peripherals: None
 ram: 3 GB
 recovery_boot: With the device powered off, hold <kbd>Home</kbd> + <kbd>Volume Up</kbd> + <kbd>Power</kbd>.
diff --git a/_data/devices/mondrianwifi.yml b/_data/devices/mondrianwifi.yml
index 5719e3a..4f47601 100644
--- a/_data/devices/mondrianwifi.yml
+++ b/_data/devices/mondrianwifi.yml
@@ -22,6 +22,7 @@ network:
 - {tech: 2G, bands: '' }
 - {tech: 3G, bands: '' }
 - {tech: 4G, bands: '' }
+no_oem_unlock_switch: true
 peripherals: None
 ram: 2 GB
 recovery_boot: With the device powered off, hold <kbd>Volume Up</kbd> while booting.
diff --git a/_data/devices/n1awifi.yml b/_data/devices/n1awifi.yml
index 5a44ce9..01b47eb 100644
--- a/_data/devices/n1awifi.yml
+++ b/_data/devices/n1awifi.yml
@@ -25,6 +25,7 @@ network:
 - {tech: 2G, bands: '' }
 - {tech: 3G, bands: '' }
 - {tech: 4G, bands: '' }
+no_oem_unlock_switch: true
 peripherals: [Accelerometer, Compass, GPS, Gyroscope]
 ram: 3 GB
 recovery_boot: With the device powered off, hold <kbd>Home</kbd> + <kbd>Volume Up</kbd> + <kbd>Power</kbd>.
diff --git a/_data/devices/n2awifi.yml b/_data/devices/n2awifi.yml
index ef854be..1ffe925 100644
--- a/_data/devices/n2awifi.yml
+++ b/_data/devices/n2awifi.yml
@@ -25,6 +25,7 @@ network:
 - {tech: 2G, bands: '' }
 - {tech: 3G, bands: '' }
 - {tech: 4G, bands: '' }
+no_oem_unlock_switch: true
 peripherals: [Accelerometer, Compass, GPS, Gyroscope]
 ram: 2 GB
 recovery_boot: With the device powered off, hold <kbd>Home</kbd> + <kbd>Volume Up</kbd> + <kbd>Power</kbd>.
diff --git a/_data/devices/n5100.yml b/_data/devices/n5100.yml
index 34cd6d2..930c879 100644
--- a/_data/devices/n5100.yml
+++ b/_data/devices/n5100.yml
@@ -25,6 +25,7 @@ network:
 - {tech: 2G, bands: '850 900 1800 1900 MHz GSM' }
 - {tech: 3G, bands: '850 900 1900 2100 MHz HSDPA' }
 - {tech: 4G, bands: '' }
+no_oem_unlock_switch: true
 peripherals: [Accelerometer, Gyroscope, Proximity sensor, Compass, GPS]
 ram: 2 GB
 recovery_boot: With the device powered off, hold <kbd>Volume Up</kbd> + <kbd>Power</kbd> until the screen flashes twice, then release both.
diff --git a/_data/devices/n5110.yml b/_data/devices/n5110.yml
index cf363d0..80f8820 100644
--- a/_data/devices/n5110.yml
+++ b/_data/devices/n5110.yml
@@ -25,6 +25,7 @@ network:
 - {tech: 2G, bands: '' }
 - {tech: 3G, bands: '' }
 - {tech: 4G, bands: '' }
+no_oem_unlock_switch: true
 peripherals: [Accelerometer, Gyroscope, Proximity sensor, Compass, GPS]
 ram: 2 GB
 recovery_boot: With the device powered off, hold <kbd>Volume Up</kbd> + <kbd>Power</kbd> until the screen flashes twice, then release both.
diff --git a/_data/devices/n5120.yml b/_data/devices/n5120.yml
index 11ddf77..eed4552 100644
--- a/_data/devices/n5120.yml
+++ b/_data/devices/n5120.yml
@@ -25,6 +25,7 @@ network:
 - {tech: 2G, bands: '850 900 1800 1900 MHz GSM' }
 - {tech: 3G, bands: '850 900 2100 MHz HSDPA' }
 - {tech: 4G, bands: '3(1800) 7(2600) 8(900) 20(800) LTE' }
+no_oem_unlock_switch: true
 peripherals: [Accelerometer, Gyroscope, Proximity sensor, Compass, GPS]
 ram: 2 GB
 recovery_boot: With the device powered off, hold <kbd>Volume Up</kbd> + <kbd>Power</kbd> until the screen flashes twice, then release both.
diff --git a/_data/devices/serrano3gxx.yml b/_data/devices/serrano3gxx.yml
index 0fb23a7..22ae51c 100644
--- a/_data/devices/serrano3gxx.yml
+++ b/_data/devices/serrano3gxx.yml
@@ -26,6 +26,7 @@ network:
 - {tech: 2G, bands: '850 900 1800 1900 MHz GSM' }
 - {tech: 3G, bands: '850 900 1900 2100 MHz HSDPA' }
 - {tech: 4G, bands: '' }
+no_oem_unlock_switch: true
 peripherals: [Accelerometer, Gyroscope, Proximity sensor, Compass, SNS integration, FM radio, Infrared sensor, GPS, GLONASS]
 ram: 1.5 GB
 recovery_boot: With the device powered off, hold <kbd>Volume Up</kbd> + <kbd>Home</kbd> + <kbd>Power</kbd>.
diff --git a/_data/devices/serranodsdd.yml b/_data/devices/serranodsdd.yml
index aeacc94..4b18e75 100644
--- a/_data/devices/serranodsdd.yml
+++ b/_data/devices/serranodsdd.yml
@@ -26,6 +26,7 @@ network:
 - {tech: 2G, bands: '850 900 1800 1900 MHz GSM' }
 - {tech: 3G, bands: '850 900 1900 2100 MHz HSDPA' }
 - {tech: 4G, bands: '' }
+no_oem_unlock_switch: true
 peripherals: [Dual SIM, Accelerometer, Gyroscope, Proximity sensor, Compass, SNS integration, FM radio, Infrared sensor, GPS, GLONASS]
 ram: 1.5 GB
 recovery_boot: With the device powered off, hold <kbd>Volume Up</kbd> + <kbd>Home</kbd> + <kbd>Power</kbd>.
diff --git a/_data/devices/serranoltexx.yml b/_data/devices/serranoltexx.yml
index f68a600..fd4551f 100644
--- a/_data/devices/serranoltexx.yml
+++ b/_data/devices/serranoltexx.yml
@@ -26,6 +26,7 @@ network:
 - {tech: 2G, bands: '850 900 1800 1900 MHz GSM' }
 - {tech: 3G, bands: '850 900 1900 2100 MHz HSDPA' }
 - {tech: 4G, bands: '3(1800) 7(2600) 8(900) 20(800)' }
+no_oem_unlock_switch: true
 peripherals: [Accelerometer, Gyroscope, Proximity sensor, Compass, SNS integration, FM radio, Infrared sensor, GPS, GLONASS, NFC]
 ram: 1.5 GB
 recovery_boot: With the device powered off, hold <kbd>Volume Up</kbd> + <kbd>Home</kbd> + <kbd>Power</kbd>.
diff --git a/_data/devices/t0lte.yml b/_data/devices/t0lte.yml
index 29ee341..3cce100 100644
--- a/_data/devices/t0lte.yml
+++ b/_data/devices/t0lte.yml
@@ -25,6 +25,7 @@ network:
 - {tech: 2G, bands: '850 900 1800 1900 MHz GSM (All models), 1800 Mhz CDMA (E250L)' }
 - {tech: 3G, bands: '850 900 1900 2100 MHz HSDPA (N7105/T), 850 1900 2100 MHz HSDPA (i317/M,T889/V), 1900 2100 Mhz HSDPA (E250S/K/L)' }
 - {tech: 4G, bands: '800 1800 2600 MHz (N7105), 1800 2600 MHz (N7105T), 700 850 1900 (i317/M), 700 (T889/V), 850 1800 Mhz (E250S), 900 1800 MHz (E250K), 850 2100 MHz (E250L)' }
+no_oem_unlock_switch: true
 peripherals: [Accelerometer, Barometer, Compass, GPS, Gyroscope, NFC, Proximity sensor, RGB ambient light]
 ram: 2 GB
 recovery_boot: With the device powered off, hold <kbd>Home</kbd> + <kbd>Volume Up</kbd> + <kbd>Power</kbd>.
diff --git a/_data/devices/v1awifi.yml b/_data/devices/v1awifi.yml
index fa4c27c..df4f36b 100644
--- a/_data/devices/v1awifi.yml
+++ b/_data/devices/v1awifi.yml
@@ -25,6 +25,7 @@ network:
 - {tech: 2G, bands: '' }
 - {tech: 3G, bands: '' }
 - {tech: 4G, bands: '' }
+no_oem_unlock_switch: true
 peripherals: [Accelerometer, Compass, GPS, Gyroscope]
 ram: 3 GB
 recovery_boot: With the device powered off, hold <kbd>Home</kbd> + <kbd>Volume Up</kbd> + <kbd>Power</kbd>.
diff --git a/_data/devices/zerofltexx.yml b/_data/devices/zerofltexx.yml
index c95082b..ebe28c3 100644
--- a/_data/devices/zerofltexx.yml
+++ b/_data/devices/zerofltexx.yml
@@ -25,6 +25,7 @@ network:
 - {tech: 2G, bands: '850 900 1800 1900 MHz GSM' }
 - {tech: 3G, bands: '850 900 1800 1900 HSDPA' }
 - {tech: 4G, bands: 'B1-B8, B12, B17-B20, B26' }
+no_oem_unlock_switch: true
 peripherals: [Fingerprint reader, Heart rate sensor, Motion Coprocessor, Accelerometer,
   Gesture sensor, Gyroscope, Proximity sensor, Compass, Barometer, Hall effect sensor,
   RGB ambient light, Infrared sensor, GPS, NFC]
diff --git a/_data/devices/zeroltexx.yml b/_data/devices/zeroltexx.yml
index d4db343..56373e0 100644
--- a/_data/devices/zeroltexx.yml
+++ b/_data/devices/zeroltexx.yml
@@ -25,6 +25,7 @@ network:
 - {tech: 2G, bands: '850 900 1800 1900 MHz GSM' }
 - {tech: 3G, bands: '850 900 1800 1900 HSDPA' }
 - {tech: 4G, bands: 'B1-B8, B12, B17-B20, B26' }
+no_oem_unlock_switch: true
 peripherals: [Fingerprint reader, Heart rate sensor, Motion Coprocessor, Accelerometer,
   Gesture sensor, Gyroscope, Proximity sensor, Compass, Barometer, Hall effect sensor,
   RGB ambient light, Infrared sensor, GPS, NFC]
diff --git a/_includes/templates/recovery_install_heimdall.md b/_includes/templates/recovery_install_heimdall.md
index 804c6f3..5c60b82 100644
--- a/_includes/templates/recovery_install_heimdall.md
+++ b/_includes/templates/recovery_install_heimdall.md
@@ -10,30 +10,33 @@ Samsung devices come with a unique boot mode called "Download mode", which is ve
 [Heimdall](http://www.glassechidna.com.au/products/heimdall/) is a cross-platform, open-source tool for interfacing with Download mode on Samsung devices.
 The preferred method of installing a custom recovery is through this boot mode{% unless custom_root_instructions %} -- rooting the stock firmware is neither necessary nor required{% endunless %}.
 
-1. Download and install the [Heimdall suite](http://glassechidna.com.au/heimdall/#downloads):
+{% unless site.data.devices[page.device].no_oem_unlock_switch %}
+1. Enable OEM unlock in the Developer options under device Settings.
+{% endunless %}
+2. Download and install the [Heimdall suite](http://glassechidna.com.au/heimdall/#downloads):
     * **Windows**: Extract the Heimdall suite and take note of the directory containing `heimdall.exe`. You can verify Heimdall is working by opening a command
     prompt in that directory and typing `heimdall version`. If you receive an error, make sure you have the 
     [Microsoft Visual C++ 2012 Redistributable Package (x86)](https://www.microsoft.com/en-us/download/details.aspx?id=30679) installed on your computer.
     * **Linux**: Pick the appropriate package to install for your distribution. The `-frontend` packages aren't needed for this guide. After installation,
     verify Heimdall is installed by running `heimdall version` in the terminal.
     * **macOS**: Install the `dmg` package. After installation, Heimdall should be available from the terminal - type `heimdall version` to double-check.
-2. Power off the device and connect the USB adapter to the computer (but not to the device, yet).
-3. Boot into download mode:
+3. Power off the device and connect the USB adapter to the computer (but not to the device, yet).
+4. Boot into download mode:
 
     * {{ site.data.devices[page.device].download_boot }}
 
     Accept the disclaimer, then insert the USB cable into the device.
-4. **Windows only**: install the drivers. A more complete set of instructions can be found in the [ZAdiag user guide](https://github.com/pbatard/libwdi/wiki/Zadig).
+5. **Windows only**: install the drivers. A more complete set of instructions can be found in the [ZAdiag user guide](https://github.com/pbatard/libwdi/wiki/Zadig).
     1. Run `zadiag.exe` from the **Drivers** folder of the Heimdall suite.
     2. Choose **Options** &raquo; **List all devices** from the menu.
     3. Select **Samsung USB Composite Device** or **MSM8x60** or **Gadget Serial** or **Device Name** from the drop down menu. (If nothing relevant appears, try uninstalling any Samsung related Windows software, like Samsung Windows drivers and/or Kies).
     4. Click **Replace Driver** (having to select **Install Driver** from the drop down list built into the button).
     5. If you are prompted with a warning that the installer is unable to verify the publisher of the driver, select **Install this driver anyway**. You may receive two more prompts about security. Select the options that allow you to carry on.
-5. On the computer, open a command prompt (on Windows) or terminal (on Linux or macOS) window, and type:
+6. On the computer, open a command prompt (on Windows) or terminal (on Linux or macOS) window, and type:
 ```
 heimdall print-pit
 ```
-6. If the device reboots, Heimdall is installed and working properly.
+7. If the device reboots, Heimdall is installed and working properly.
 
 {% if custom_downgrade_instructions %}
 {{ custom_downgrade_instructions }}
-- 
2.17.1

